webpack/webpack5-localization-plugin/src/AssetProcessor.ts

```diff
@@ -228,19 +229,25 @@ function _reconstructLocalized(
 
         const escapedBackslash: string = element.escapedBackslash || '\\';
 
-        // Replace backslashes with the properly escaped backslash
-        BACKSLASH_REGEX.lastIndex = -1;
-        newValue = newValue.replace(BACKSLASH_REGEX, escapedBackslash);
-
-        // @todo: look into using JSON.parse(...) to get the escaping characters
-        const escapingCharacterSequence: string = escapedBackslash.slice(escapedBackslash.length / 2);
+        if (newValue.includes('\\')) {
+          // The vast majority of localized strings do not contain `\\`, so this check avoids an allocation.
+          // Replace backslashes with the properly escaped backslash
+          BACKSLASH_REGEX.lastIndex = -1;
+          newValue = newValue.replace(BACKSLASH_REGEX, escapedBackslash);
+        }
 
         // Ensure the the quotemark, apostrophe, tab, and newline characters are properly escaped
         ESCAPE_REGEX.lastIndex = -1;
-        newValue = newValue.replace(
-          ESCAPE_REGEX,
-          (match) => `${escapingCharacterSequence}${ESCAPE_MAP.get(match)}`
-        );
+        if (ESCAPE_REGEX.test(newValue)) {
+          // The majority of localized strings do not contain the characters that need to be escaped,
+          // so this check avoids an allocation.
+          // @todo: look into using JSON.parse(...) to get the escaping characters
+          const escapingCharacterSequence: string = escapedBackslash.slice(escapedBackslash.length / 2);
```

Comment:
Is this math required each time? Won't all escaped backslashes be of length 2? (Could be wrong... just not familiar of any other escape scheme that would be larger)

Reply:
Reverted.